ctapipe_io_magic/laser_params_corr_test.py

```diff
@@ -0,0 +1,411 @@
+        gkBGSamplesV1to4 = None
+        with open(filename, 'r') as file:
+            for line in file:
+                line = line.strip()
+                if line and not line.startswith("#"):  # Skip empty lines and comments
+                    variable, value = line.split("=")
+                    variable = variable.strip()
+                    value = value.split("#")[0].strip()  # Remove comments
+                    if variable == "gkBGSamplesV1to4":
+                        try:
+                            gkBGSamplesV1to4 = int(value)
+                        except ValueError:
+                            raise ValueError("Invalid value for gkBGSamplesV1to4 in params.txt")
+
+        if gkBGSamplesV1to4 is None:
+            raise ValueError("Variable gkBGSamplesV1to4 not found in params.txt")
+
+        if not isinstance(bgsamples, int):
+            raise ValueError("Invalid type for bgsamples")
+
+        if bgsamples <= gkBGSamplesV1to4:
+            return 1.0
+        if ifadc < 128:
+            return 0.5
+        if ifadc < 256:
+            return 1.0
+        if ifadc < 384:
+            return 2.0
+        if ifadc < 448:
+            return 4.0
+        return 4.0 * ncollapse
+
+    def get_offset_r(self, ifadc, bgsamples, ncollapse):
+        filename = '/home/zywuckan/ctapipe_io_magic/ctapipe_io_magic/params.txt'
+        gkBGSamplesV1to4 = None
+        gkIntegrationWindow = None
+        with open(filename, 'r') as file:
+            for line in file:
+                exec(line.strip())
+                if line.strip():  # Check if line is not empty
+                    variable, value = line.split("=")
+                    variable = variable.strip()
+                    value = value.split("#")[0].strip()  # Remove comments
+                    if variable == "gkBGSamplesV1to4":
+                        try:
+                            gkBGSamplesV1to4 = int(value)
+                        except ValueError:
+                            raise ValueError("Invalid value for gkBGSamplesV1to4 in params.txt")
+                    elif variable == "gkIntegrationWindow":  # Fix indentation here
+                        try:
+                            gkIntegrationWindow = float(value)
+                        except ValueError:
+                            raise ValueError("Invalid value for gkIntegrationWindow in params.txt")
+
+        if gkBGSamplesV1to4 is None:
+            raise ValueError("Variable gkBGSamplesV1to4 not found in params.txt")
+
+        if gkIntegrationWindow is None:
+            raise ValueError("Variable gkIntegrationWindow not found in params.txt")
+
+        if not isinstance(bgsamples, int):
+            raise ValueError("Invalid type for bgsamples")
+
+        if bgsamples <= gkBGSamplesV1to4:
+            return 0.0
+        if ifadc < 128:
+            return - bgsamples * gkIntegrationWindow * self.get_bin_width(ifadc, bgsamples-1, ncollapse)
```

Comment:
I hope this function will not be executed for every event, because it is doing file operations